src/IO/ADIOS/ADIOS1IOHandler.cpp

```diff
@@ -37,31 +95,527 @@ ADIOS1IOHandler::flush()
+
+        /* Every variable (i.e. dataset for our purposes) write in ADIOS noxml style requires an individual ID.
+         * Creating a dataset explicitly at this point is not necessary. */
+
+        m_datasetSize[writable] = getBP1Extent(parameters.extent);
+
+        writable->written = true;
+        writable->abstractFilePosition = std::make_shared< ADIOS1FilePosition >(name);
+
+        auto res = m_filePath.find(writable);
+        if( res == m_filePath.end() )
+            res = m_filePath.find(writable->parent);
+
+        m_filePath[writable] = res->second;
+    }
+}
+
+void
+ADIOS1IOHandlerImpl::extendDataset(Writable* writable,
+                                   Parameter< Operation::EXTEND_DATASET > const& parameters)
 { }
 
-ADIOS1IOHandler::~ADIOS1IOHandler()
+void
+ADIOS1IOHandlerImpl::openFile(Writable* writable,
+                              Parameter< Operation::OPEN_FILE > const& parameters)
 { }
 
-std::future< void >
-ADIOS1IOHandler::flush()
+void
+ADIOS1IOHandlerImpl::openPath(Writable* writable,
+                              Parameter< Operation::OPEN_PATH > const& parameters)
+{ }
+
+void
+ADIOS1IOHandlerImpl::openDataset(Writable* writable,
+                                 Parameter< Operation::OPEN_DATASET >& parameters)
+{ }
+
+void
+ADIOS1IOHandlerImpl::deleteFile(Writable* writable,
+                                Parameter< Operation::DELETE_FILE > const& parameters)
+{ }
+
+void
+ADIOS1IOHandlerImpl::deletePath(Writable* writable,
+                                Parameter< Operation::DELETE_PATH > const& parameters)
+{ }
+
+void
+ADIOS1IOHandlerImpl::deleteDataset(Writable* writable,
+                                   Parameter< Operation::DELETE_DATASET > const& parameters)
+{ }
+
+void
+ADIOS1IOHandlerImpl::deleteAttribute(Writable* writable,
+                                     Parameter< Operation::DELETE_ATT > const& parameters)
+{ }
+
+void
+ADIOS1IOHandlerImpl::writeDataset(Writable* writable,
+                                  Parameter< Operation::WRITE_DATASET > const& parameters)
 {
-    return m_impl->flush();
+    if( m_handler->accessType == AccessType::READ_ONLY )
+        throw std::runtime_error("Writing into a dataset in a file opened as read only is not possible.");
+
+    std::string name = concrete_bp1_file_position(writable);
+
+    ADIOS_DATATYPES datatype = getBP1DataType(parameters.dtype);
+
+    std::string dims = getBP1Extent(parameters.extent);
+    std::string const& global_dims = m_datasetSize.at(writable);
+    std::string local_offsets = getBP1Extent(parameters.offset);
+
+    int64_t id;
+    id = adios_define_var(m_group,
+                          name.c_str(),
+                          "",
+                          datatype,
+                          dims.c_str(),
+                          global_dims.c_str(),
+                          local_offsets.c_str());
+    ASSERT(id >= 0 /* ??? */, "Internal error: Failed to define ADIOS variable during Dataset writing");
```

Comment:
good candidate for a `VERIFY` macro - seen this a lot in the last on startup/env issues.

Reply:
Very much agreed. We should not exclude these assertions in production code, unless the user *really* wants to.